static/js/apps/biomed_nl/app.tsx

```diff
@@ -86,23 +102,125 @@ export function App(): ReactElement {
+        "The data is taking its time to allow for complete osmosis of information.",
+        "The data is making sure all its enzymes are perfectly aligned for the reaction.",
+      ],
+      []
+    );
+    // Show puns in random order every time.
+    const shuffledIndexes = useRef(
+      [...Array(customMessages.length).keys()].sort(() => Math.random() - 0.5)
+    ).current;
+
+    useEffect(() => {
+      // Show first biomed pun after 3 seconds
+      const timeoutId = setTimeout(() => {
+        setShowMessage(true);
+        setMessage(customMessages[shuffledIndexes[messageIndex]]);
+      }, 3000);
+
+      // Show next pun after one minute
+      const messageInterval = setInterval(() => {
+        setMessageIndex((prevIndex) => (prevIndex + 1) % customMessages.length);
+        setMessage(customMessages[shuffledIndexes[messageIndex]]);
+      }, 60000);
+
+      return () => {
+        clearTimeout(timeoutId);
+        clearInterval(messageInterval);
+      };
+    }, [customMessages, messageIndex, shuffledIndexes]);
+
+    return (
+      <>
+        <div className="spinner-container">
+          <div className="spinner"></div>
+          <div>This may take up to 5 minutes...</div>
+        </div>
+        {showMessage && (
+          <div
+            css={css`
+              margin: ${theme.spacing.xl}px;
+            `}
+          >
+            <p
+              css={css`
+                ${theme.typography.text.md};
+              `}
+            >
+              {message}*
+            </p>
+            <p
+              css={css`
+                ${theme.typography.text.sm};
+              `}
+            >
+              * Puns generated by Gemini
+            </p>
+          </div>
+        )}
+      </>
+    );
+  }
+
   /**
    * useEffect hook to handle initial loading of information from the URL hash.
    */
   useEffect(() => {
-    const hashParams = queryString.parse(window.location.hash);
-    const hashQuery = (hashParams[URL_HASH_PARAMS.q] || "") as string;
-    if (hashQuery) {
-      setQuery(hashQuery);
+    const handleHashChange = (): void => {
+      const hashParams = queryString.parse(window.location.hash);
+      const hashQuery = (hashParams[URL_HASH_PARAMS.q] || "") as string;
+      if (hashQuery) {
+        onHashChange(hashQuery);
+      }
+    };
+
+    // Execute query when hash params are updated.
+    window.addEventListener("hashchange", handleHashChange);
+
+    // Check if query is provided in hash param on load.
+    handleHashChange();
+
+    // Cleanup the event listener
+    return () => {
+      window.removeEventListener("hashchange", handleHashChange);
+    };
+  }, []); // Run only once to set up the listener
+
+  function setHashToQuery(): void {
+    updateHash({ [URL_HASH_PARAMS.q]: query });
+  }
+
+  function handleKeydownEvent(
+    event: React.KeyboardEvent<HTMLDivElement>
+  ): void {
+    // Navigate through the suggested results.
```

Comment:
what is this comment referring to?

Reply:
left overs from copying and pasting 🫣 -- removed